src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1988,7 +1988,8 @@ private static string CheckForMatches(uint[] runningHash, int upTo)
                     upTo = upToMax;
                 }
 
-                for (var i = 0; i < upTo; i++)
+                // Skip all before shortest suspicious string. Today it is 'Emit' so start with i=3.
+                for (var i = MIN_LEN - 1; i < upTo; i++)
```

Comment:
By looking at the hashing algorithm used in `Match`, we should just only check `upTo - 1` here, no need to do the loop if the passed-in `upTo` is not 0. The loop is needed only when the passed in `upTo` is 0.